addons/mail/static/src/core/common/thread.js

```diff
@@ -178,6 +179,11 @@ export class Thread extends Component {
                 toRaw(nextProps.thread).fetchNewMessages();
             }
         });
+        onRendered(() => {
+            if (this.env.inDiscussApp) {
+                this.env.config.setDisplayName(this.props.thread.displayName);
+            }
+        });
```

Comment:
I think this should be in discuss component directly so we don't need this condition